ios/RNCPushNotificationIOS.m

```diff
@@ -315,7 +315,9 @@ - (void)handleRemoteNotificationRegistrationError:(NSNotification *)notification
     if (error != NULL) {
       reject(@"-1", @"Error - Push authorization request failed.", error);
     } else {
-      [RCTSharedApplication() registerForRemoteNotifications];
+      dispatch_async(dispatch_get_main_queue(), ^(void){
+        [RCTSharedApplication() registerForRemoteNotifications];
+      });
       [UNUserNotificationCenter.currentNotificationCenter getNotificationSettingsWithCompletionHandler:^(UNNotificationSettings * _Nonnull settings) {
```

Comment:
You should run the getNotificationSettingsWithCompletionHandler: call into Main Thread as well.

Reply:
My goal was to change the minimum amount to clear Apple's warnings. They do not show an warning about calling that function from another thread, so I didn't change it. This PR is simply to clean up Main Thread warnings, so this suggested addition is out of scope.

Reply:
Yep but as you get it this code was running in background thread so the getNotificationSettingsWithCompletionHandler: will be too even if you didn't get warning by Apple is good pratice in iOS to register observer in main thread.
By experience we faced issues where when you don't register on main thread your observer completion was never fired.

Reply:
I disagree. You shouldn't be doing more work on the Main (UI) thread than you have to. If Apple is happy to let you grab those settings from a non-UI thread, you should do so. The Main thread should be reserved for actions that require user interaction or drawing of elements.